2-mandatory/2-retrying-random-numbers.js

```diff
@@ -5,20 +5,26 @@
 
 // This function shouldn't be changed
 function generateRandomNumber() {
-    console.log("Generating number...");
-    return Math.round(Math.random() * 100);
+  console.log("Generating number...");
+  return Math.round(Math.random() * 100);
 }
 
 function getRandomNumberGreaterThan50() {
-    // TODO - implement using a do-while loop
+  let number = 0;
+  do {
+    return (number = getRandomNumberGreaterThan50());
```

Comment:
There is a little bit of a problem with this function, you're calling the function you are within, rather than calling `generateRandomNumber()`

So if we run through it...
Number is equal to zero
Whilst number is less than or equal to 50, call the function which defines number as zero and assign that as number.

It's an infinite loop, unless I am missing something.